src/main/java/nu/studer/teamcity/buildscan/internal/cleanup/CustomDataStorageBuildScanCleaner.java

```diff
@@ -0,0 +1,88 @@
+package nu.studer.teamcity.buildscan.internal.cleanup;
+
+import jetbrains.buildServer.serverSide.SBuildServer;
+import jetbrains.buildServer.serverSide.SQLRunner;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static nu.studer.teamcity.buildscan.internal.CustomDataStorageBuildScanDataStore.BUILD_SCAN_STORAGE_ID;
+
+@SuppressWarnings("deprecation")
+public final class CustomDataStorageBuildScanCleaner {
+
+    private static final int BATCH_SIZE = 10;
```

Comment:
This seems a very low number, though might very well be justified. Can you please elaborate?

Reply:
The majority of rows to be deleted are in the subtable `custom_data_body` - this is just grouping 10 build ids together 1 batch. The actual effected rows in the subtable is very varying, that's why I came up with a defensive low number.

Reply:
👍